include/nuttx/can/sja1000.h

```diff
@@ -0,0 +1,236 @@
+/****************************************************************************
+ * include/nuttx/can/sja1000.h
+ *
+ * SJA1000 CAN driver based on esp32c3_twai.h
+ *
+ * License header retained from original source.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __INCLUDE_NUTTX_CAN_SJA1000_H
+#define __INCLUDE_NUTTX_CAN_SJA1000_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Types
+ ****************************************************************************/
+
+/* CAN hardware-dependent bit-timing constant
+ *
+ * Used for calculating and checking bit-timing parameters
+ */
+
+struct can_bittiming_const_s
+{
+  char name[16];      /* Name of the CAN controller hardware */
```

Comment:
Why to include the CAN controller name? Do you want to print it to the user?

Reply:
It is equivalent to my original ScketCAN proposal. It has next reason, if in (Linux) kernel bitrate computation is not used, trusted and some external application takes information over IOCTL, does certified parameters computation and then sets raw bittiming parameters back to the chip, then some carmakers has specific certified combinations for each controller used in the car network. The field allows to check that actual enumerated interfaces matches one of these rules.

If  can_bittiming_const_s instance for specific controller is defined as const, then even for NuttX RAM constrained targets it is not a problem. But I personally do not insist on this field for NuttX. On the other hand 1:1 match to Linux is good.